dags/veda_data_pipeline/veda_dataset_pipeline.py

```diff
@@ -50,29 +49,9 @@
     start = EmptyOperator(task_id="start")
     end = EmptyOperator(task_id="end")
 
-
-    @task()
-    def remove_thumbnail_asset(ti):
-        payload = deepcopy(ti.dag_run.conf)
-        payloads = list()
-        assets = payload.get("assets", {})
-        if assets.get("thumbnail"):
-            assets.pop("thumbnail")
-        # if thumbnail was only asset, delete assets
-        if not assets:
-            payload.pop("assets")
-        for item in payload.get("discovery_items"):
-            payloads.append({
-                **payload,
-                **item
-            }
-            )
-
-        return payloads
-
-
     mutated_payloads = start >> collection_task_group() >> remove_thumbnail_asset()
-    discover = discover_from_s3_task.expand(payload=mutated_payloads)
+    discovery_items = extract_discovery_items_from_payload(mutated_payloads)
```

Comment:
The function `extract_discovery_items_from_payload` doesn't need any input since it uses task_instance (ti) to get the configuration. We can simply do
```python
discovery_items = extract_discovery_items_from_payload()
```
And Airflow will use its magic to define the `ti` needed by `extract_discovery_items_from_payload` task